tests/backup/backup_helper.go

```diff
@@ -1443,3 +1452,207 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		log.Infof("%s job not found", pxCentralPostInstallHookJobName)
+	}
+
+	// Get storage class using for px-backup deployment
+	statefulSet, err := apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNamespace)
+	if err != nil {
+		return err
+	}
+	pvcs, err := apps.Instance().GetPVCsForStatefulSet(statefulSet)
+	if err != nil {
+		return err
+	}
+	storageClassName := pvcs.Items[0].Spec.StorageClassName
+
+	// Get the tarball required for helm upgrade
+	cmd = fmt.Sprintf("curl -O  https://raw.githubusercontent.com/portworx/helm/%s/stable/px-central-%s.tgz", latestPxBackupHelmBranch, versionToUpgrade)
+	log.Infof("curl command to get tarball: %v ", cmd)
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("error downloading of tarball: %v", err)
+	}
+	log.Infof("Terminal output: %s", output)
+
+	// Execute helm upgrade using cmd
+	log.Infof("Upgrading Px-Backup version from %s to %s", currentBackupVersionString, versionToUpgrade)
+	cmd = fmt.Sprintf("helm upgrade px-central px-central-%s.tgz --namespace %s --create-namespace --version %s --set persistentStorage.enabled=true,persistentStorage.storageClassName=\"%s\",pxbackup.enabled=true",
+		versionToUpgrade, pxBackupNamespace, versionToUpgrade, *storageClassName)
+	log.Infof("helm command: %v ", cmd)
+	output, _, err = osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("upgrade failed with error: %v", err)
+	}
+	log.Infof("Terminal output: %s", output)
+
+	// Wait for post install hook job to be completed
+	postInstallHookJobCompletedCheck := func() (interface{}, bool, error) {
+		job, err := batch.Instance().GetJob(pxCentralPostInstallHookJobName, pxBackupNamespace)
+		if err != nil {
+			return "", true, err
+		}
+		if job.Status.Succeeded > 0 {
+			log.Infof("Status of job %s after completion - "+
+				"\nactive count - %d"+
+				"\nsucceeded count - %d"+
+				"\nfailed count - %d\n", job.Name, job.Status.Active, job.Status.Succeeded, job.Status.Failed)
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("status of job %s not yet in desired state - "+
+			"\nactive count - %d"+
+			"\nsucceeded count - %d"+
+			"\nfailed count - %d\n", job.Name, job.Status.Active, job.Status.Succeeded, job.Status.Failed)
+	}
+	_, err = task.DoRetryWithTimeout(postInstallHookJobCompletedCheck, 10*time.Minute, 30*time.Second)
+	if err != nil {
+		return err
+	}
+
+	// Checking if all pods are running
+	allPods, err := core.Instance().GetPods(pxBackupNamespace, nil)
+	log.Infof("Listing all the pods in Px Backup Namespace [%s] - ", pxBackupNamespace)
+	for _, pod := range allPods.Items {
+		log.Infof(pod.GetName())
+	}
+	if err != nil {
+		return err
+	}
+	for _, pod := range allPods.Items {
+		log.Infof("Checking status for pod - %s", pod.GetName())
+		err = core.Instance().ValidatePod(&pod, 5*time.Minute, 30*time.Second)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Shouldn't error check be done first?
And maybe combine the two loops? I'm not sure why they are two separate loops.
I mentioned it in https://github.com/portworx/torpedo/pull/1237#discussion_r1146042879, and you said you "fixed it". does that mean this is how it _should_ be?

Reply:
Fixed it. We did not need the first loop since we are printing the name of the pods in the second loop anyway.